static/js/stream_edit.js

```diff
@@ -701,6 +720,30 @@ export function initialize() {
         }
         stream_ui_updates.update_regular_sub_settings(sub);
 
+        const topic = narrow_state.topic();
+        // Refresh the view only if the stream is unsubscribed before
+        // and subscribed now
+        if (!sub.subscribed) {
+            const opened_sub = narrow_state.stream_sub();
+            // The opened stream is not the same as the selected stream to update,
+            // then do not need to refresh the message view of the former.
+            if (opened_sub === undefined || opened_sub.stream_id !== sub.stream_id) {
+                return;
+            }
+            if (topic === undefined) {
+                narrow.activate([{operator: "stream", operand: sub.name}], {
+                    trigger: "subscribe button",
+                });
+            } else {
+                narrow.activate(
+                    [
+                        {operator: "stream", operand: sub.name},
+                        {operator: "topic", operand: topic},
+                    ],
+                    {trigger: "subscribe button"},
+                );
```

Comment:
I think we can avoid duplication of code here. Also, I feel it would be better to reload the view once a confirmation is received from the server and thus we can do so after receiving the subscription event. So, I think we should add code to handle this in `mark_subscribed` and this would also help us in avoiding duplication of code.